agent/task.go

```diff
@@ -149,7 +149,7 @@ func (a *Agent) runPreTaskCommands(ctx context.Context, tc *taskContext) {
 			return
 		}
 		if taskGroup.SetupGroup != nil {
-			err = a.runCommands(ctx, tc, taskGroup.SetupGroup.List(), false)
+			err = a.runCommands(ctx, tc, taskGroup.SetupGroup.List(), taskGroup.SetupGroupFailTask)
 			if err != nil {
```

Comment:
Okay I changed it so that this function only returns an error in the SetupGroup case, let me know if we want to cancel early for other situations as well.